TFNetworkRecLayer.py

```diff
@@ -2834,21 +2868,22 @@ def get_loop_acc_layer(name):
       with tf.name_scope(self.layer_data_templates[name].layer_class_type.cls_get_tf_scope_name(name)):
         inner_layer = self.net.get_layer(name)
         acc_ta = loop_accumulated["output_%s" % name]
-        acc_ta, search_choices = self._opt_search_resolve(
-          layer_name=name, acc_ta=acc_ta, final_net_vars=final_net_vars)
+        acc_ta, search_choices, resolved_seq_len = self._opt_search_resolve(
+          layer_name=name, acc_ta=acc_ta, final_net_vars=final_net_vars, seq_len=seq_len)
         output = self.layer_data_templates[name].output.copy_template_adding_time_dim(time_dim_axis=0)
         output.beam = search_choices.get_beam_info() if search_choices else None
+        max_len = tf.reduce_max(resolved_seq_len)
         # We should have accumulated it.
         output.placeholder = tensor_array_stack(acc_ta, stop=max_len)  # e.g. (time,batch,dim)
-        output.size_placeholder = {0: seq_len}
+        output.size_placeholder = {0: resolved_seq_len}
         if output.beam:
           if self.parent_rec_layer.output.beam:
             # seq_len should have already a beam, same as rec_layer.output
             assert output.beam == self.parent_rec_layer.output.beam
           else:
-            output.size_placeholder[0] = tile_transposed(seq_len, axis=0, multiples=output.beam.beam_size)
-            if time_dim_tag:
-              time_dim_tag.set_tag_on_size_tensor(output.size_placeholder[0])
+            output.size_placeholder[0] = tile_transposed(resolved_seq_len, axis=0, multiples=output.beam.beam_size)
+        if time_dim_tag:
+          time_dim_tag.set_tag_on_size_tensor(output.size_placeholder[0])
```

Comment:
I was missing this change to set the tag again after a new seq_len tensor is created. Now tests are passing.